packages/svelte/src/containers/xy-container/xy-container.svelte

```diff
@@ -46,26 +48,39 @@
   })
 
   setContext('component', () => ({
-    update: (c: XYComponentCore<Datum>) => { config.components = [...config.components, c] },
+    update: (c: XYComponentCore<Datum>) => {
+      if (config.components?.some((e) => e === c)) return
```

Comment:
We can probably use the `includes()` function here instead

Reply:
fixed